### PR TITLE
Home page course

### PR DESCRIPTION
**Pull request by salman0719**
- Inserting course primary filter into home page banner
  - Feature flag (`FEATURE_ENABLE_COURSE_SEARCH` environment variable) needs to be enabled (set to `1`).
- Adjusting Banner & Badge appearance styles for better responsiveness
- Cache revalidation issue workaround for next@14.0.3

### CODE REVIEW

**Review by mikemike27 — COMMENTED** [review]
All good!

File: src/app/components/banner/banner.module.css [hunk 1]
```diff
@@ -9,61 +11,41 @@
...
+
+  .form {
+    display: none !important;
   }
-  .mobileSearchBar{
-   display: block;
-   height: 40vh;
+
+  .formMobile {
+    display: flex !important;
```
Comment (bhumkarpratik) [inline src/app/components/banner/banner.module.css:49]: Do we need to use `!important` here? Can we not use this?

File: src/app/components/banner/banner.module.css [hunk 2]
```diff
@@ -9,61 +11,41 @@
...
+
+  .bannerTextAndBadge {
     flex-direction: column;
-    justify-content: center;
   }
-  .descktopSearchBar{
-    display: none;
+
+  .form {
+    display: none !important;
```
Comment (bhumkarpratik) [inline src/app/components/banner/banner.module.css:45]: Same

**Review by bhumkarpratik — APPROVED** [review]
LGTM

**Pull request merged**